BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/advertising/AdvertisingPacketUtil.java

```diff
@@ -29,12 +29,31 @@ public static UUID toUuid(byte[] bytes) {
         return rssis;
     }
 
-    public static float getMeanRssi(int[] rssis) {
-        int rssiSum = 0;
-        for (int i = 0; i < rssis.length; i++) {
-            rssiSum += rssis[i];
+    public static float calculateMean(int[] values) {
+        int sum = 0;
+        for (int i = 0; i < values.length; i++) {
+            sum += values[i];
         }
-        return rssiSum / (float) rssis.length;
+        return sum / (float) values.length;
+    }
+
+    public static float calculateVariance(int[] values) {
+        float average = calculateMean(values);
+
+        float squaredDistanceSum = 0;
+        for (int i = 0; i < values.length; i++) {
+            squaredDistanceSum += Math.pow(values[i] - average, 2);
+        }
+        int sampleLength = Math.max(values.length - 1, 1);
+        return squaredDistanceSum / sampleLength;
+    }
+
+    public static float getPacketFrequency(int packets, long time) {
```

Comment:
`(time / 1000)` will produce `java.lang.ArithmeticException: / by zero` for `time < 1000`. Change `time == 0` to `time < 1000` or use `1000f`. Later one may create huge frequency values for small time intervals.